src/modules/0.0.22/aws_s3/rpcs/s3_clean_bucket.ts

```diff
@@ -38,29 +45,58 @@ export class S3CleanBucketRpc extends RpcBase {
         },
       ];
     }
-    const client = (await ctx.getAwsClient()) as AWS;
-    const objects = await this.getBucketObjects(client.s3Client, bucketName);
-    for (const object of objects) {
-      // delete the object
-      await this.deleteBucketObject(client.s3Client, bucketName, object.Key);
-    }
+    const client = (await ctx.getAwsClient(await ctx.getDefaultRegion())) as AWS;
 
-    // query again to see if all objects have been deleted
-    const remainingObjects = await this.getBucketObjects(client.s3Client, bucketName);
-    if (!remainingObjects.length) {
-      return [
-        {
-          bucket: bucketName,
-          status: 'OK',
-          response_message: 'All bucket objects have been deleted',
-        },
-      ];
+    // first determine bucket region
+    let region = await this.getBucketLocation(client.s3Client, bucketName);
+    if (!region) region = 'us-east-1';
```

Comment:
Can make this const by making it a one-liner.

```suggestion
    const region = (await this.getBucketLocation(client.s3Client, bucketName)) ?? 'us-east-1';
```